frontend/app/_components/client-dashboard-components/Header.js

```diff
@@ -15,7 +39,7 @@ function Header() {
         <LogOutButton />
       </div>
       <div className="flex justify-between items-end text-primary-800 text-3xl px-10 h-40">
-        <p>Welcome Rose</p>
+        <p>{error ? error : `Welcome ${userName}`}</p>
```

Comment:
in general it is bad practice to output the error message to the use. It is good to notify the user of the errors, but in a way it makes sense for them - either in a call for action (refresh the page) or information (we couln't find your user... sing up or double check credentials) something like that :)

also, I believe the error is never set `setError` function is never used, or am I missing something?